### DISCUSSION TITLE
Regarding Menubar list

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#684] [General Discussion]
Hello Team,

As we're currently on release 5.0
I wanted to bring to your attention an issue I've encountered in the menu bar listing of our current release, 5.0. 
Specifically, when I read the menu bar form, only the home tab appears as "true." Interestingly, even after logging in, I am still able to see only the home tab. 
I have attached the screenshort when i logging in for the first time.
![Screenshot from 2024-02-08 11-14-34](https://github.com/Sunbird-Ed/Community/assets/72926048/dda020ba-aa83-41ca-8885-665695ff3015)

However, upon refreshing the browser page, all the tab menus become visible.
Here is the screenshort for the menu after refresh.
![Screenshot from 2024-02-08 11-14-11](https://github.com/Sunbird-Ed/Community/assets/72926048/17ce062b-0b06-4857-a419-8800d3be8c81)

I'm reaching out to see if anyone has encountered a similar issue or might have insights on how to address it. 
Any suggestions or guidance on how to fix this would be greatly appreciated.

@vinukumar-vs @rajeevsathish @rajeshkumaravel @Krishnaj20

**Comment by rajeevsathish** [comment]
@siddhishinde0723 

what is the need? do you need to see always only home tab. Then please update the form with only one in it and delete the other tabs.

cc: @vinukumar-vs

**Reply by siddhishinde0723** [reply]
@rajeevsathish   Okay, I'll try doing that

**Reply by siddhishinde0723** [reply]
@rajeevsathish I tried updating the form with only one in it and delete the other tabs still getting the same issue after refresh all the tabs are visible.

**Comment by rajeevsathish** [comment]
@siddhishinde0723 can you please give us the curl and the response once and then we can have a look on the form updated as I think the form is cached in the browser cache

**Reply by siddhishinde0723** [reply]
@rajeevsathish here is the curl of the forms

This is the read curl:
`curl --location '{{host}}/api/data/v1/form/read' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "type": "contentcategory",
        "action": "menubar",
        "subType": "global",
        "rootOrgId": ""
    }
}'`

This is the create curl:
`curl --location '{{host}}/api/data/v1/form/create' \
--header 'Authorization: Bearer ' \
--header 'Content-Type: Application/json' \
--data '{
    "request": {
      "type": "contentcategory",
        "action": "menubar",
        "subType": "global",
        "rootOrgId": "",
        "framework": "*",
         "data": {
                "templateName": "menuConfig",
                "action": "list",
                "fields": [
                    {
                        "index": 0,
                        "title": "frmelmnts.lbl.home",
                        "desc": "frmelmnts.lbl.home",
                        "menuType": "Content",
                        "contentType": "home",
                        "isEnabled": true,
                        "default": true,
                        "isOnlineOnly": true,
                        "theme": {
                            "baseColor": "",
                            "textColor": "",
                            "supportingColor": "",
                            "className": "home",
                            "imageName": "courses-banner-img.svg"
                        },
                        "anonumousUserRoute": {
                            "route": "/explore",
                            "queryParam": "home"
                        },
                        "loggedInUserRoute": {
                            "route": "/resources",
                            "queryParam": "home"
                        },
                        "metaData": {
                            "filters": [
                                "board",
                                "gradeLevel",
                                "subject",
                                "medium",
                                "publisher",
                                "audience",
                                "channel"
                            ],
                            "groupByKey": "subject",
                            "defaultFilters": {
                                "board": [
                                    "CBSE"
                                ],
                                "gradeLevel": [
                                    "Class 10"
                                ],
                                "medium": []
                            }
                        },
                        "search": {
                            "facets": [
                                "subject",
                                "primaryCategory",
                                "medium",
                                "banner",
                                "additionalCategories",
                                "search",
                                "ContinueLearning"
                            ],
                            "fields": [
                                "name",
                                "appIcon",
                                "medium",
                                "subject",
                                "resourceType",
                                "contentType",
                                "organisation",
                                "topic",
                                "mimeType",
                                "trackable",
                                "gradeLevel",
                                "se_boards",
                                "se_subjects",
                                "se_mediums",
                                "se_gradeLevels"
                            ],
                            "filters": {
                                "channel": [],
                                "subject": [],
                                "audience": [],
                                "primaryCategory": [
                                    "Digital Textbook",
                                    "eTextbook",
                                    "Course"
                                ],
                                "board": [],
                                "medium": [],
                                "gradeLevel": []
                            }
                        },
                        "filter": {
                            "isEnabled": false,
                            "type": "facet"
                        },
                        "sections": [
                            {
                                "index": 4,
                                "title": "frmelmnts.lbl.recentlyPublishedCourses",
                                "desc": "Section for Searched contents",
                                "facetKey": "search",
                                "isEnabled": true,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {
                                        "request": {
                                            "filters": {
                                                "contentType": [
                                                    "Course"
                                                ],
                                                "primaryCategory": [
                                                    "Course"
                                                ],
                                                "batches.enrollmentType": "open",
                                                "batches.status": [
                                                    1
                                                ],
                                                "status": [
                                                    "Live"
                                                ],
                                                "se_boards": [
                                                    "State (Tamil Nadu)"
                                                ],
                                                "se_mediums": [
                                                    "English",
                                                    "Tamil"
                                                ],
                                                "se_gradeLevels": [
                                                    "Class 1",
                                                    "Class 2",
                                                    "Class 5"
                                                ]
                                            },
                                            "limit": 100,
                                            "fields": [
                                                "name",
                                                "appIcon",
                                                "mimeType",
                                                "gradeLevel",
                                                "identifier",
                                                "medium",
                                                "pkgVersion",
                                                "board",
                                                "subject",
                                                "resourceType",
                                                "contentType",
                                                "channel",
                                                "organisation",
                                                "trackable",
                                                "se_boards",
                                                "se_subjects",
                                                "se_mediums",
                                                "se_gradeLevels",
                                                "lastPublishedOn"
                                            ],
                                            "facets": [
                                                "se_subjects"
                                            ],
                                            "sort_by": {
                                                "lastPublishedOn": "desc"
                                            }
                                        }
                                    },
                                    "params": "",
                                    "contextKey": "res.facet.search",
                                    "sortBy": "lastPublishedOn"
                                }
                            },
                            {
                                "index": 0,
                                "title": "Continue Learning",
                                "desc": "Continue Learning",
                                "facetKey": "ContinueLearning",
                                "isEnabled": true
                            },
                            {
                                "index": 5,
                                "title": "frmelmnts.lbl.fromncert",
                                "defaultTitle": "From CBSE",
                                "desc": "Section for Searched contents",
                                "facetKey": "search",
                                "isEnabled": true,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {
                                        "request": {
                                            "filters": {
                                                "primaryCategory": [
                                                    "Course",
                                                    "Digital Textbook"
                                                ],
                                                "status": [
                                                    "Live"
                                                ],
                                                "se_boards": [
                                                    "cbse"
                                                ],
                                                "channel": "0124487522476933120"
                                            },
                                            "limit": 100,
                                            "fields": [
                                                "name",
                                                "appIcon",
                                                "mimeType",
                                                "gradeLevel",
                                                "identifier",
                                                "medium",
                                                "pkgVersion",
                                                "board",
                                                "subject",
                                                "resourceType",
                                                "contentType",
                                                "channel",
                                                "organisation",
                                                "trackable",
                                                "se_boards",
                                                "se_subjects",
                                                "se_mediums",
                                                "se_gradeLevels",
                                                "lastPublishedOn"
                                            ],
                                            "facets": [
                                                "se_subjects"
                                            ],
                                            "sort_by": {
                                                "lastPublishedOn": "desc"
                                            }
                                        }
                                    },
                                    "params": "",
                                    "contextKey": "res.facet.search",
                                    "sortBy": "lastPublishedOn"
                                }
                            },
                            {
                                "index": 3,
                                "title": "frmelmnts.lbl.search",
                                "desc": "Section for Searched contents",
                                "facetKey": "search",
                                "isEnabled": true,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {
                                        "request": {
                                            "filters": {
                                                "channel": [],
                                                "subject": [],
                                                "audience": [],
                                                "primaryCategory": [
                                                    "Digital Textbook"
                                                ],
                                                "se_boards": [
                                                    "State (Tamil Nadu)"
                                                ],
                                                "se_mediums": [
                                                    "English"
                                                ],
                                                "se_gradeLevels": [
                                                    "Class 4"
                                                ]
                                            },
                                            "limit": 100,
                                            "fields": [
                                                "name",
                                                "appIcon",
                                                "mimeType",
                                                "gradeLevel",
                                                "identifier",
                                                "medium",
                                                "pkgVersion",
                                                "board",
                                                "subject",
                                                "resourceType",
                                                "primaryCategory",
                                                "contentType",
                                                "channel",
                                                "organisation",
                                                "trackable",
                                                "se_boards",
                                                "se_subjects",
                                                "se_mediums",
                                                "se_gradeLevels"
                                            ],
                                            "facets": [
                                                "se_subjects"
                                            ]
                                        }
                                    },
                                    "params": "",
                                    "contextKey": "res.facet.search"
                                }
                            },
                            {
                                "index": 2,
                                "title": "frmelmnts.lbl.bannerTitle",
                                "desc": "Section for Banner",
                                "facetKey": "banner",
                                "isEnabled": true
                            },
                            {
                                "index": 1,
                                "title": "frmelmnts.lbl.subjects",
                                "desc": "Section for subjects",
                                "facetKey": "subject",
                                "isEnabled": true,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {},
                                    "params": "",
                                    "contextKey": "res.facet.subject"
                                },
                                "theme": {
                                    "component": "sb-pills-grid",
                                    "limit": 10,
                                    "infiniteCard": false,
                                    "colorMapping": [
                                        {
                                            "primary": "rgba(255,139,46,1)",
                                            "secondary": "rgba(255,139,46,0.3)"
                                        },
                                        {
                                            "primary": "rgba(163,99,255,1)",
                                            "secondary": "rgba(163,99,255,0.3)"
                                        },
                                        {
                                            "primary": "rgba(34,139,255,1)",
                                            "secondary": "rgba(34,139,255,0.3)"
                                        },
                                        {
                                            "primary": "rgba(95,192,32,1)",
                                            "secondary": "rgba(95,192,32,0.3)"
                                        },
                                        {
                                            "primary": "rgba(255,128,47,1)",
                                            "secondary": "rgba(255,128,47,0.3)"
                                        },
                                        {
                                            "primary": "rgba(251,70,70,1)",
                                            "secondary": "rgba(251,70,70,0.3)"
                                        },
                                        {
                                            "primary": "rgba(83,109,252,1)",
                                            "secondary": "rgba(83,109,252,0.3)"
                                        },
                                        {
                                            "primary": "rgba(15,186,208,1)",
                                            "secondary": "rgba(15,186,208,0.3)"
                                        }
                                    ],
                                    "icons": {
                                        "english": "assets/images/book_english.svg",
                                        "mathematics": "assets/images/calculator.svg",
                                        "science": "assets/images/globe.svg",
                                        "default": "assets/images/book_default.svg"
                                    }
                                },
                                "landing": {
                                    "title": "frmelmnts.lbl.exploretitle",
                                    "description": "frmelmnts.lbl.exploredescription"
                                },
                                "anonumousUserRoute": {
                                    "route": "/explore-course",
                                    "queryParam": "course"
                                },
                                "loggedInUserRoute": {
                                    "route": "/learn",
                                    "queryParam": "course"
                                }
                            },
                            {
                                "index": 6,
                                "title": "frmelmnts.lbl.dscfrmctgries",
                                "desc": "Section for category",
                                "facetKey": "primaryCategory",
                                "isEnabled": true,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {},
                                    "params": "",
                                    "contextKey": "res.facet.primaryCategory"
                                },
                                "filter": {
                                    "primaryCategory": [
                                        "course",
                                        "digital textbook",
                                        "etextbook"
                                    ],
                                    "additionalCategories": [
                                        "tv lesson"
                                    ]
                                },
                                "merge": {
                                    "source": "primaryCategory",
                                    "destination": "additionalCategories"
                                },
                                "theme": {
                                    "component": "sb-pills-grid",
                                    "limit": 10,
                                    "infiniteCard": false,
                                    "colorMapping": [
                                        {
                                            "primary": "rgba(255,255,255,1)",
                                            "secondary": "rgba(255,255,255,1)"
                                        }
                                    ],
                                    "icons": {
                                        "digital textbooks": "assets/images/textbook.svg",
                                        "courses": "assets/images/course.svg",
                                        "tv programs": "assets/images/tv.svg",
                                        "documents": "assets/images/documents.svg",
                                        "videos": "assets/images/videos.svg",
                                        "default": "assets/images/all_content.svg"
                                    }
                                },
                                "landing": {
                                    "title": "frmelmnts.lbl.exploretitle",
                                    "description": "frmelmnts.lbl.exploredescription"
                                },
                                "anonumousUserRoute": {
                                    "route": "/explore-course",
                                    "queryParam": "course"
                                },
                                "loggedInUserRoute": {
                                    "route": "/learn",
                                    "queryParam": "course"
                                }
                            },
                            {
                                "index": 7,
                                "title": "frmelmnts.lbl.medium",
                                "desc": "Section for medium",
                                "facetKey": "medium",
                                "isEnabled": false,
                                "apiConfig": {
                                    "url": "",
                                    "method": "",
                                    "req": {},
                                    "params": "",
                                    "contextKey": "res.facet.medium"
                                },
                                "theme": {
                                    "component": "sb-pills-grid",
                                    "limit": 10,
                                    "infiniteCard": false,
                                    "colorMapping": [
                                        {
                                            "primary": "rgba(255,139,46,1)",
                                            "secondary": "rgba(255,139,46,0.3)"
                                        },
                                        {
                                            "primary": "rgba(163,99,255,1)",
                                            "secondary": "rgba(163,99,255,0.3)"
                                        },
                                        {
                                            "primary": "rgba(34,139,255,1)",
                                            "secondary": "rgba(34,139,255,0.3)"
                                        },
                                        {
                                            "primary": "rgba(95,192,32,1)",
                                            "secondary": "rgba(95,192,32,0.3)"
                                        },
                                        {
                                            "primary": "rgba(255,128,47,1)",
                                            "secondary": "rgba(255,128,47,0.3)"
                                        },
                                        {
                                            "primary": "rgba(251,70,70,1)",
                                            "secondary": "rgba(251,70,70,0.3)"
                                        },
                                        {
                                            "primary": "rgba(83,109,252,1)",
                                            "secondary": "rgba(83,109,252,0.3)"
                                        },
                                        {
                                            "primary": "rgba(15,186,208,1)",
                                            "secondary": "rgba(15,186,208,0.3)"
                                        }
                                    ]
                                },
                                "anonumousUserRoute": {
                                    "route": "/home",
                                    "queryParam": "course"
                                },
                                "loggedInUserRoute": {
                                    "route": "/learn",
                                    "queryParam": "course"
                                }
                            }
                        ]
                    }
                ]
            }
    }
}'`

**Comment by rajeevsathish** [comment]
@siddhishinde0723 looks good but only question is do you have any form that is for
curl --location '{{host}}/api/data/v1/form/read' \ --header 'Content-Type: application/json' \ --data '{ "request": { "type": "contentcategory", "action": "menubar", "subType": "global", "rootOrgId": "*" , "framework": "*" } }' or any form that is specific for the rootorgId that you have created 

please check and update

**Reply by siddhishinde0723** [reply]
@rajeevsathish the same curl is user to create the form for specific rootorgId.

**Comment by rajeevsathish** [comment]
@siddhishinde0723 is the host path shareable if so please share the host so that I can have a look into the UI and the issue

**Reply by siddhishinde0723** [reply]
@rajeevsathish here is the curl 
curl --location 'https://sunbirdsaas.com//api/data/v1/form/read' \ --header 'Content-Type: application/json' \ --data '{ "request": { "type": "contentcategory", "action": "menubar", "subType": "global", "rootOrgId": "*" } }'

**Comment by rajeevsathish** [comment]
@siddhishinde0723 I can see there are multiple forms that is why when you refresh the form with rootOrgId * is coming please make sure that the forms are unique

**Reply by siddhishinde0723** [reply]
@rajeevsathish I tried keeping one form and updating still getting facing the same issue.

**Comment by anisha147** [comment]
Hi @siddhishinde0723 

Could you let me know if you were able to resolve the issue? If not, we can schedule a call to resolve the issues quickly.

**Reply by siddhishinde0723** [reply]
Hi @anisha147 
This problem has been resolved. I appreciate the advice.
